extensions/metadata_exchange/plugin.cc

```diff
@@ -88,34 +88,20 @@ void PluginRootContext::updateMetadataValue() {
       Base64::encode(metadata_bytes.data(), metadata_bytes.size());
 }
 
-// onConfigure == false makes the proxy crash.
-// Only policy plugins should return false.
+// Metadata exchange has sane defaults and therefore it will be fully
+// functional even with configuration errors.
+// A configuration error thrown here will cause the proxy to crash.
 bool PluginRootContext::onConfigure(size_t size) {
-  initialized_ = configure(size);
-  return true;
-}
-
-bool PluginRootContext::configure(size_t) {
   updateMetadataValue();
   if (!getValue({"node", "id"}, &node_id_)) {
-    logDebug("cannot get node ID");
+    LOG_DEBUG("cannot get node ID");
   }
-  logDebug(absl::StrCat("metadata_value_ id:", id(), " value:", metadata_value_,
-                        " node:", node_id_));
+  LOG_DEBUG(absl::StrCat("metadata_value_ id:", id(),
+                         " value:", metadata_value_, " node:", node_id_));
 
   // Parse configuration JSON string.
-  std::unique_ptr<WasmData> configuration = getConfiguration();
-  auto j = ::Wasm::Common::JsonParse(configuration->view());
-  if (!j.is_object()) {
-    logWarn(absl::StrCat("cannot parse plugin configuration JSON string: ",
-                         configuration->view(), j.dump()));
-    return false;
-  }
-
-  auto max_peer_cache_size =
-      ::Wasm::Common::JsonGetField<int64_t>(j, "max_peer_cache_size");
-  if (max_peer_cache_size.has_value()) {
-    max_peer_cache_size_ = max_peer_cache_size.value();
+  if (!configure(size)) {
```

Comment:
We can make it `if (size > 0 && configure(size))`, so that this log won't even show when user does not specify anything.

Reply:
updated. PTAL.